### PR TITLE
Convert Delegation Pattern to Completion Handler

### PR DESCRIPTION
**Pull request by KunJeongPark**
### Summary of changes

- convert from returning results via delegate functions to merchants to returning results via completion handlers.
- provide async await version of public functions

### Checklist

- [x] Added a changelog entry

- @KunJeongPark

### CODE REVIEW

File: Demo/Demo/ViewModels/CardPaymentViewModel.swift [hunk 1]
```diff
@@ -117,11 +117,22 @@ class CardPaymentViewModel: ObservableObject, CardDelegate {
...
             cardClient = CardClient(config: config)
             payPalDataCollector = PayPalDataCollector(config: config)
-            cardClient?.delegate = self
             let cardRequest = CardRequest(orderID: orderID, card: card, sca: sca)
-            cardClient?.approveOrder(request: cardRequest)
+            cardClient?.approveOrder(request: cardRequest) { result, error in
+                if let error {
+                    self.setUpdateSetupTokenFailureResult(vaultError: error)
+                } else if let result {
+                    self.approveResultSuccessResult(
```
Comment (sshropshire) [inline Demo/Demo/ViewModels/CardPaymentViewModel.swift:125]: qq: should this method be named `approveOrderSuccessResult()`?

Comment (KunJeongPark) [reply Demo/Demo/ViewModels/CardPaymentViewModel.swift:125]: That's a good call. There are many changes I want to make on the demo app, but I stuck to ones relevant to the conversion. 
Like needing the loading progressView where I forgot them, to see if there were delays introduced in PayPal vault.

**Comment by sshropshire** [comment]
I do see how completion handlers could be useful though. We do have some time in the beta to figure out the best strategy. Ultimately offering both sounds like a viable option to me.

**Comment by KunJeongPark** [comment]
> I do see how completion handlers could be useful though. We do have some time in the beta to figure out the best strategy. Ultimately offering both sounds like a viable option to me.

- yes, with both PRs, there will be async/await and completion handler versions of the public functions.
  It's just a matter of which set is wrapped.

**Review by Nikki1980 — APPROVED** [review]

**Review by sshropshire — APPROVED** [review]

File: Demo/Demo/SwiftUIComponents/PayPalVaultViews/PayPalVaultView.swift [hunk 2]
```diff
@@ -15,13 +15,18 @@ struct PayPalVaultView: View {
                     )
                     SetupTokenResultView(vaultViewModel: paypalVaultViewModel)
                     if let setupTokenID = paypalVaultViewModel.state.setupToken?.id {
-                        Button("Vault PayPal") {
-                            Task {
-                                await paypalVaultViewModel.vault(setupTokenID: setupTokenID)
+                        ZStack {
+                            Button("Vault PayPal") {
+                                Task {
```
Comment (sshropshire) [inline Demo/Demo/SwiftUIComponents/PayPalVaultViews/PayPalVaultView.swift:20]: random q: would it be cleaner if we created the `Task` within the `paypalVaultViewModel.vault()` method?

Comment (KunJeongPark) [reply Demo/Demo/SwiftUIComponents/PayPalVaultViews/PayPalVaultView.swift:20]: Button actions are synchronous contexts, so I need to create a Task block there to call an async/await function.

Comment (KunJeongPark) [reply Demo/Demo/SwiftUIComponents/PayPalVaultViews/PayPalVaultView.swift:20]: It makes sense what you are saying, make the views look cleaner. Let me add that to demo app refactor ticket.

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,9 +5,32 @@
...
+      * Add `start(request:) async throws -> PayPalCheckoutResult`
+      * Add `vault(vaultRequest:) async throws -> PayPalVaultResult`
+  * CardPayments
+    * Replace delegate pattern with completion handlers and Swift concurrency
+      * Remove `CardDelegate` and `CardVaultDelegate`
+      * Remove `approveOrder(request:)` method that uses delegate callbacks
+      * Remove `vault(vaultRequest:)` method that uses delegate callbacks
+      * Add `approveOrder(request:completion:(CardResult?, Error?) -> Void)` to `CardClient`
+      * Add `vault(request:completion:(CardVaultResult?, Error?) -> Void)` to `CardClient`
+      * Add `approveOrder(request:) async throws -> Cardresult`
```
Comment (sshropshire) [inline CHANGELOG.md]:
```suggestion
      * Add `approveOrder(request:) async throws -> CardResult`
```

Comment (KunJeongPark) [reply CHANGELOG.md]: Thank you! Nice catch!

**Pull request merged**